src/components/video/SimpleVideo.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+import { Animated, StyleSheet, View, ViewStyle } from 'react-native';
+// @ts-ignore
+import Sound from 'react-native-sound';
+// @ts-ignore
+import Video from 'react-native-video';
+import convertToProxyURL from 'react-native-video-cache';
+import styled from 'styled-components';
+import { useTheme } from '../../context/ThemeContext';
+import { useAudio } from '@rainbow-me/hooks';
+import { ImgixImage } from '@rainbow-me/images';
+import { position } from '@rainbow-me/styles';
+import logger from 'logger';
+
+export type SimpleVideoProps = {
+  readonly style?: ViewStyle;
+  readonly uri: string;
+  readonly posterUri?: string;
+  readonly loading: boolean;
+  readonly setLoading: (isLoading: boolean) => void;
+};
+
+const absoluteFill = `
+  position: absolute;
+  ${position.size('100%')}
+`;
+
+const StyledBackground = styled(View)`
+  ${absoluteFill}
+`;
+
+const StyledVideo = styled(Video)`
+  ${absoluteFill}
+`;
+
+const StyledPosterContainer = styled(Animated.View)`
+  ${absoluteFill}
+`;
+
+const StyledImgixImage = styled(ImgixImage)`
+  ${position.size('100%')}
+`;
+
+const styles = StyleSheet.create({
+  flex: { flex: 1 },
+});
+
+export default function SimpleVideo({
+  style,
+  uri,
+  posterUri,
+  loading,
+  setLoading,
+}: SimpleVideoProps): JSX.Element {
+  const { colors } = useTheme();
+  const { white } = colors;
+  const ref = React.useRef<Video>();
+  const source = React.useMemo(
+    () => ({
+      uri: convertToProxyURL(uri),
+    }),
+    [uri]
+  );
+  const { currentSound, isPlayingAsset, fadeTo } = useAudio();
+  const [opacity] = React.useState<Animated.Value>(
+    () => new Animated.Value(loading ? 1 : 0)
```

Comment:
It is not a good pattern to this. It is the opposite sense of `Animated` API. 
Please, use `useEffect` and `setValue`. It won't lead to any updates on JS side.

Reply:
This only happens when the component is mounted. We're achieving the same as `useMemo` except that we can make it sensitive to the props on mount only (otherwise it would reallocate on every prop change).

Reply:
we have `useAnimatedValue` for that

Reply:
@osdnk I can see `useAnimatedValue` in `src/components/WelcomeScreen` and `useAnimatedListener` which isn't used at all. Do you want me to move `useAnimatedValue` to a shared hook?